datadog/resource_datadog_dashboard.go

```diff
@@ -5622,25 +5653,234 @@ func buildTerraformHostmapRequestStyle(datadogStyle datadogV1.HostMapWidgetDefin
 }
 
 // Schema validation
+func validateWidgetPalette(val interface{}, key string) (warns []string, errs []error) {
+	_, err := datadogV1.NewWidgetPaletteFromValue(val.(string))
+	if err != nil {
```

Comment:
With a little bit of elbow grease, I feel that we could define a wrapper function that you can directly assign to `ValidateFunc`. Seems non trivial. In the mean time, please inline the err check like `if _, err := ..; err != nil`